### PR TITLE
Function to generate Flow types from edit data

### PR DESCRIPTION
**Pull request by mwiencek**
A SQL function, `edit_data_type_info()`, is added that generates "type info" for a single column of edit data.

A script, script/generate_edit_data_flow_type.js, is added which takes an edit type, `selects edit_data_type_info()` for every row of edit data for that type, and outputs a single merged Flow type representing the edit data overall for that type.

For example:

```sh
./script/generate_edit_data_flow_type.js --edit-type 92
```

Generates a Flow type for `EDIT_RELATIONSHIP_DELETE`.

The `PROD_STANDBY` database must be configured in DBDefs.pm to use this utility.

The underlying code used to build the Flow type may also be useful to generate Flow types for other JSON data, e.g. entities.json, so I've added script/generate_json_flow_type.js for this more general purpose too. Here's an example invoking that script:

```sh
cat entities.json | jq -c | script/generate_json_flow_type.js
```

(This expects the JSON object on a single line, so `jq -c` is used to compact the data first.)

I've committed the type generated for root/edit/details/RemoveRelationship.js as an example.

I think this will help with two things:
1. The SQL function by itself is useful to see all the different data formats an edit type has, which can make it easier to identify places we can clean up the data later.
2. The combined Flow type generated by edit_data_flow_type.js is useful for accessing the edit data safely (type-checked by Flow) in a React template. This isn't so important now, but in the future we'll likely want to move `build_display_data` etc. into JS.

### CODE REVIEW

**Comment by mwiencek** [comment]
Note that we can use script/edit_data_flow_type.js to generate types for other JSON too, e.g. entities.json. I'd like to do that in another branch once this is merged. :)

**Review by reosarevok — APPROVED** [review]
LGTM but I didn't test

**Review by yvanzo — APPROVED** [review]
Successfully tested on production database with “Reorder relationships” edits (type ID = 99).

Would it make sense to have a wrapper script that uses `PROD_STANDBY` from `DBDefs`?

That would be nice to mention the command generating the added flow type in second commit too.

**Comment by mwiencek** [comment]
> Would it make sense to have a wrapper script that uses PROD_STANDBY from DBDefs?

Added a wrapper script for that (and updated the commit message/PR description with instructions). It sets `statement_timeout = 0` so you shouldn't get timeouts anymore.

Also added `script/generate_json_flow_type.js` to generate a Flow type from other JSON.

File: package.json
Comment (mwiencek) [inline package.json:52]: I'm also using these in my json-dumps rework.

**Review by yvanzo — CHANGES_REQUESTED** [review]
```bash
./script/generate_edit_data_flow_type.js --edit-type 92
```
returned:

```node
/musicbrainz-server/root/utility/generateFlowType.js:37
      (isEditDataTypeInfo ?? parent.isEditDataTypeInfo);
                           ^

SyntaxError: Unexpected token ?
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/musicbrainz-server/script/generate_edit_data_flow_type.js:23:28)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
```

using Node `v10.19.0` currently installed in MusicBrainz Docker.

**Review by yvanzo — APPROVED** [review]
Works fine now. Code looks good too. Maybe document scripts also in `HACKING.md`?

**Comment by mwiencek** [comment]
> Maybe document scripts also in HACKING.md?

I wasn't sure where to put that so I made a general Flow section. Have a quick look. :)

File: HACKING.md
Comment (yvanzo) [inline HACKING.md:162]:
```suggestion
If you're adding a new JS file, it should have a `@flow strict-local` header at the top
```

File: HACKING.md
Comment (yvanzo) [inline HACKING.md:184]: Not sure how useful that will be though?

**Review by yvanzo — APPROVED** [review]
Each time I approve this pull request, there are new additions, so I approve it again!

File: HACKING.md
Comment (mwiencek) [reply HACKING.md:184]: It seems more generally useful to me than the one for edit data. You commonly might want to generate a type for some request data (say, JSON from a web service endpoint like AB/CB/AcoustID). Whereas the edit data one is really only useful for core developers writing edit code.

**Comment by mwiencek** [comment]
OK, I'll commit the `strict-local` addition and merge this to save you from another approval. :)

**Pull request merged**
